src/parser/jobs/rdm/modules/MeleeCombos.js

```diff
@@ -0,0 +1,562 @@
+				verstone: this.combatants.selected.hasStatus(STATUSES.VERSTONE_READY.id),
+				verfire: this.combatants.selected.hasStatus(STATUSES.VERFIRE_READY.id),
+				acceleration: this.combatants.selected.hasStatus(STATUSES.ACCELERATION.id),
+			},
+			events: [event],
+			lastAction: event,
+		}
+	}
+
+	_breakComboIfExists() {
+		if (this._currentCombo) {
+			this._currentCombo.broken = true
+			this._endCombo()
+		}
+	}
+
+	_endCombo() {
+		this._meleeCombos[this._currentCombo.start] = this._currentCombo
+		delete this._currentCombo
+	}
+
+	_handleFinisher() {
+		const isFireReady = this.combatants.selected.hasStatus(STATUSES.VERFIRE_READY.id)
+		const isStoneReady = this.combatants.selected.hasStatus(STATUSES.VERSTONE_READY.id)
+		const combo = this._currentCombo
+		combo.finisher = {
+			recommendedActions: [],
+			recommendation: '',
+		}
+		const finisherUsed = combo.events[combo.events.length-1].ability
+
+		let recommendedFinisher = null
+		if (combo.startmana.white < combo.startmana.black) {
+			recommendedFinisher = this._outOfBalanceFinisher(combo.startmana.white, isStoneReady, WHITE_MANA_ACTIONS, combo.startmana.black, isFireReady, BLACK_MANA_ACTIONS)
+		} else if (combo.startmana.black < combo.startmana.white) {
+			recommendedFinisher = this._outOfBalanceFinisher(combo.startmana.black, isFireReady, BLACK_MANA_ACTIONS, combo.startmana.white, isStoneReady, WHITE_MANA_ACTIONS)
+		} else {
+			recommendedFinisher = this._inBalanceFinisher(combo.startmana.black, isFireReady, BLACK_MANA_ACTIONS, combo.startmana.white, isStoneReady, WHITE_MANA_ACTIONS)
+		}
+
+		if (recommendedFinisher instanceof Array) {
+			if (recommendedFinisher === FINISHERS) {
+				// a recommendation of both finishers means ignore the finisher, either one is valid
+				combo.finisher.recommendedActions.push(getAction(finisherUsed.guid))
+			} else {
+				// a recommendation of an array of actions is to delay the combo
+				Array.prototype.push.apply(combo.finisher.recommendedActions, recommendedFinisher)
+				this._incorrectFinishers.delay++
+				combo.finisher.recommendation = <Trans id="rdm.meleecombos.recommendation.delay">Delay your melee combo to guarantee a proc from your finisher, if you will lose 7 or less mana to overcapping.</Trans>
+			}
+		} else {
+			// A specific finisher was recommended
+			combo.finisher.recommendedActions.push(recommendedFinisher)
+			if (finisherUsed.guid !== recommendedFinisher.id) {
+				// wrong finisher was used, add an incorrect finisher tally
+				if (finisherUsed.guid === ACTIONS.VERHOLY.id) {
+					this._incorrectFinishers.verholy++
+				}
+				if (finisherUsed.guid === ACTIONS.VERFLARE.id) {
+					this._incorrectFinishers.verflare++
+				}
+				combo.finisher.recommendation = <Trans id="rdm.meleecombos.recommendation.incorrect">See the suggestions section for finisher guidelines.</Trans>
+			}
+		}
+	}
+
+	_outOfBalanceFinisher(lowerMana, lowerManaProc, lowerManaActions, higherMana, higherManaProc, higherManaActions) {
```

Comment:
few more args than i'm comfy with - could probably clean it up as like
```js
// Only need to do this once for each mana type
const whiteState = {
	amount: ...,
	proc: ...,
	actions: ...,
}

// Then can pass directly
this._outOfBalanceFinisher(whiteState, blackState) // or (blackState, whiteState)
```
can destructure the params on the receiving end

Reply:
Implemented.  I'm not sure I like the readability of this method versus the original, but it's not a huge difference either way.